### PR TITLE
Refresh button

### PR DESCRIPTION
**Pull request by mjustian**
### What does this change intend to accomplish?
There was no easy way to reset a broken stream before. This adds a button to the player view and the modal in settings to do that. The buttons interface with a stream command on the REST API that resets a stream. IT supports both BaseStream and PersistentStream.

### Checklist

* [X] Have you tested your changes and ensured they work?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] If applicable, have you updated the documentation/manual?
* [X] If applicable, have you updated the CHANGELOG?
* [X] Does your submission pass linting & tests? You can test on localhost using `./scripts/test`
* [x] If this is a UI change, have you tested it across multiple browser platforms?
* [X] If this is a UI change, have you tested across multiple viewport sizes (ie. desktop versus mobile)?

Web View:
reset0

iPad:
reset_ipad

iPhone 12 Mini:
reset_iphone12mini

iPhone SE
reset_iphonese

Stream Modal:
reset_modal

### CODE REVIEW

**Review by rtertiaer — CHANGES_REQUESTED** [review]
if you absolutely must wholesale run files thru a linter, can you at least put those in separate commits so I can easily review what's functionally changed? (not a huge fan of that anyhow, unless you're substantially rewriting a file anyhow.)

**Comment by linknum23** [comment]
Opinions below. I'll let others chime in with their thoughts and see if we can reach a consensus.

- I really like being able to reset the stream from the stream modal since it takes a couple of clicks to get there and has an explanation as to what it does and why.
- My instinct is that the actual reset button on the player is too easy and takes up too much space for an operation that shouldn't be needed under most scenarios.
- After talking with Michael on this I like the word restart for this instead of reset. I think it is a less ambiguous name and clearly explains what it does in one work.

**Comment by gorski123** [comment]
The restart button should be shown by the currently playing stream. I dont think we need to write out reset or restart, rather just a button with a logo (reset logo or the current recycle logo you are using) should suffice. I was thinking something like this:
image
image

**Comment by linknum23** [comment]
I don't think the main view with all of the streams needs a reset button.
It doesn't even have the much more used play/pause/next song button.

**Comment by mjustian** [comment]
If the main view had a button like that, it would be really hard to tap on a mobile device.

**Comment by mjustian** [comment]
Also, would the end user know the recycle symbol would mean "restart stream"? Next to the icon it isn't really clear that it's a button.

**Comment by gorski123** [comment]
Hows about just putting it next to all the other buttons (play pause etc)? Thats prolly where it belongs...

**Comment by mjustian** [comment]
That would make sense, but on views such as Pandora which has 5 buttons already, it would be a bit long, especially on a mobile device. I'm not 100% sure if that would be worth making the rest of the buttons much smaller to fit the reset button.

**Comment by gorski123** [comment]
Can the buttons wrap to a second line if they dont fit? The pandora buttons already go off the screen in mobile, perhaps implementing some code to wrap to a second line could squash that bug as well.

**Comment by mjustian** [comment]
I think it would look odd if the thumbs up and thumbs down buttons are on different levels. Also, putting it with the rest of the controls conflicts with the "having a restart button so easily accessible means it will break often" opinion. It might be better to make the controls smaller on mobile, rather than wrapping to a second line.

**Review by rtertiaer — APPROVED** [review]
one last request, but otherwise you have my approval. could you detail in a comment in the code why you need a `src = self.src` in `BaseStream`'s `restart()`, but not in `PersistentStream`?

**Comment by rtertiaer** [comment]
oh, and rebasing + squashing this is very preferred too.

**Comment by mjustian** [comment]
Reason for src = self.src in BaseStream:
the _disconnect function sets self.src to None, but it is also needed to run the connect function. The PersistentStream does not have any arguments when activating, so it's not needed.

**Pull request merged**
